ansible/roles/test/files/ptftests/hash_test.py

```diff
@@ -85,11 +85,15 @@ def check_hash(self, hash_key):
 
         hit_count_map = {}
         for _ in range(0, self.BALANCING_TEST_TIMES):
+            in_port = random.choice([port for port in self.src_ports if port not in exp_port_list]) if hash_key == "ingress-port" else in_port
             logging.info("in_port: {}".format(in_port))
             (matched_index, _) = self.check_ip_route(hash_key, in_port, dst_ip, exp_port_list)
             hit_count_map[matched_index] = hit_count_map.get(matched_index, 0) + 1
         logging.info("hit count map: {}".format(hit_count_map))
-        self.check_balancing(next_hop.get_next_hop(), hit_count_map)
```

Comment:
in loose mode, you do not need to enter the check_balancing function. you can just check if the hit_count_map has two hits after line 91, if yes, then you return true. this would be working for all hash fields, and it can save lots of time.

Reply:
done, but do not use for all hash fields. Because it will weaken the test results.